src/qiboml/models/optimizers.py

```diff
@@ -0,0 +1,391 @@
+from qibo.models.encodings import hamming_weight_encoder
+from qibo import get_backend
+
+
+class ExactGeodesicTransportCG:
+    """Exact Geodesic Transport with Conjugate Gradients Optimizer.
+
+    Implements the Exact Geodesic Transport with Conjugate Gradients (EGT-CG) optimizer, a curvature-aware Riemannian optimizer designed
+    specifically for variational circuits based on the Hamming-weight encoder (HWE) ansätze. It updates parameters along exact geodesic paths
+    on the hyperspherical manifold defined by the HWE, combining analytic metric computation, conjugate-gradient memory, and dynamic
+    learning rates for fast, globally convergent optimization.
+
+    Args:
+      n_qubits (int): Number of qubits in the quantum circuit.
```

Comment:
```suggestion
      nqubits (int): Number of qubits in the quantum circuit.
```

Here and everywhere else.